Library/Homebrew/rubocops/bottle_block_cop.rb

```diff
@@ -1,20 +1,31 @@
+require_relative "./extend/formula_cop"
+
 module RuboCop
   module Cop
     module Homebrew
-      class CorrectBottleBlock < Cop
-        MSG = "Use rebuild instead of revision in bottle block".freeze
+      # This cop audits `bottle` block in Formulae
+      #
+      # - `rebuild` should be used instead of `revision` in `bottle` block
 
-        def on_block(node)
-          return if block_length(node).zero?
-          method, _args, body = *node
-          _keyword, method_name = *method
+      class CorrectBottleBlock < FormulaCop
+        MSG = "Use rebuild instead of revision in bottle block".freeze
 
-          return unless method_name == :bottle
-          check_revision?(body)
+        def on_formula(_node, _class_node, _parent_class_node, body)
+          check(body)
         end
 
         private
 
+        def check(body)
+          body.each_child_node(:block) do |block_node|
```

Comment:
`body.each_child_node(:block)` calls the given block for each `child` node which has `child.type==:block`. (We want to iterate only over blocks and find `bottle` block). 
The given block for `body.each_child_node(:block)` takes `block_node` as the argument.